llvm/include/llvm/CodeGen/SelectionDAG.h

```diff
@@ -2111,6 +2115,10 @@ class SelectionDAG {
   /// Test whether the given SDValue is known to contain non-zero value(s).
   bool isKnownNeverZero(SDValue Op, unsigned Depth = 0) const;
 
+  /// Test whether the given float value is known to not be negative. 0.0 is
+  /// considered non-negative, -0.0 is considered negative.
+  bool isKnownNonNegativeFP(SDValue Op) const;
```

Comment:
Someday we'll have to reinvent computeKnownFPClass in the DAG... (maybe we could do this in CGP to avoid that)